src/main/java/io/leonis/example/ControllerExample.java

```diff
@@ -0,0 +1,83 @@
+package io.leonis.example;
+
+import com.google.common.collect.*;
+import io.leonis.game.engine.ControllerStrategySupplier;
+import io.leonis.ipc.CliSettings;
+import io.leonis.subra.game.data.Player.PlayerIdentity;
+import io.leonis.subra.game.data.*;
+import io.leonis.subra.ipc.network.StrategyMulticastSubscriber;
+import io.leonis.subra.ipc.peripheral.*;
+import io.leonis.subra.ipc.peripheral.JamepadController.JamepadControllerIdentity;
+import io.leonis.zosma.ipc.ip.MulticastSubscriber;
+import io.leonis.zosma.ipc.peripheral.Controller.MappingSupplier;
+import java.io.IOException;
+import java.net.InetAddress;
+import java.util.AbstractMap.SimpleImmutableEntry;
+import java.util.*;
+import java.util.Map.Entry;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+import reactor.core.publisher.Flux;
+
+/**
+ * The Class ControllerExample.
+ *
+ * @author Jeroen de Jong
+ */
+public class ControllerExample {
+
+  private final static Map<String, String> DEFAULTS = ImmutableMap.of(
+      "port", "10001",
+      "ip", "224.0.0.1");
+
+  /**
+   * Constructs a new ControllerExample which submits {@link io.leonis.subra.protocol.Robot generated
+   * commands} to multicast on the supplied ip and port.
+   *
+   * @param ip The IP of the multicast destination as a {@link String}
+   * @param port The port of the multicast destination as an integer.
+   */
+  public ControllerExample(
+      final String ip,
+      final int port,
+      final Map<JamepadControllerIdentity, Set<PlayerIdentity>> controllerMapping
+  ) throws IOException {
+    // the controller handler which parses the active controls
+    final Function<JamepadController, PlayerCommand> handler = new JamepadControllerHandler();
+
+    // create a stream of gamepad states based on the mapping
+    Flux.from(new JamepadPublisher(controllerMapping))
+        .map(MappingSupplier::getAgentMapping)
+        // for each controller,
```

Comment:
i think this comment was going somewhere 😛

Reply:
you removed it :')